test/acceptance/application.test.ts

```diff
@@ -19,375 +20,82 @@ describe('Controllers', () => {
-      let flux: Todo;
-      before(async () => {
-        shopping = (await client.post('/todo').send({
-          title: 'go shopping'
-        })).body;
-        flux = (await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'flux capacitor'
-        })).body;
-      });
+  it('gets a todo by ID', async () => {
+    const todo = await givenTodo(payload as Todo);
+    delete todo.checklist;
+    await client
+      .get(`/todo/${todo.id}`)
+      .send()
+      .expect(200, todo);
+  });
 
-      it('deletes an item if it exists', async () => {
-        await client
-          .del(`/todo/${shopping.id}/items/${flux.id}`)
-          .send()
-          .expect(200);
-      });
+  it('replaces the todo by ID', async () => {
+    const todo = await givenTodo(payload as Todo);
+    await client
+      .put(`/todo/${todo.id}`)
+      .send({
+        title: 'another two'
+      })
+      .expect(200);
+  });
 
-      it('returns 404 if no match is found', async () => {
-        await client
-          .del(`/todo/${shopping.id}/items/999999`)
-          .send()
-          .expect(404);
-      });
-    });
+  it('updates the todo by ID ', async () => {
+    const todo = await givenTodo(payload as Todo);
+    await client
+      .patch(`/todo/${todo.id}`)
+      .send({
+        title: 'another three'
+      })
+      .expect(200);
+  });
 
-    describe('deleteAll', () => {
-      let shopping: Todo;
-      
-      before(async () => {
-        shopping = (await client.post('/todo').send({
-          title: 'go shopping'
-        })).body;
-        await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'flux capacitor'
-        });
-        await client.post(`/todo/${shopping.id}/items`).send({
-          title: 'plutonium'
-        });
-      });
-      it('deletes all items if they exist', async () => {
-        const path = `/todo/${shopping.id}/items`;
-        await client
-          .del(path)
-          .send()
-          .expect(200);
-        // Should not have items anymore.
-        await client
-          .get(path)
-          .send()
-          .expect(200, []);
-      });
-    });
+  it('deletes the todo', async () => {
+    const todo = await givenTodo(payload as Todo);
+    await client
+      .del(`/todo/${todo.id}`)
+      .send()
+      .expect(200);
   });
 
   function givenAnApplication() {
     app = new TodoApplication({
       rest: {
         port: 0
       },
-      datasources: {
-        ds: {
-          name: 'ds',
-          connector: 'memory'
-        }
+      datasource: {
+        connector: 'memory'
       }
     });
   }
 
   async function givenARestServer() {
     server = await app.getServer(RestServer);
   }
+
+  /**
+   * Insert a todo into the database (use for test prep)
+   * @param todo The todo to insert into the database
+   */
+  async function givenTodo(todo: Todo) {
+    const repo = (await app.get('repositories.todo')) as TodoRepository;
```

Comment:
Can we perhaps move this to a new application method?

```ts
export class TodoApplication extends Application {
  // ...
  
  async getTodoRepository() {
    return (await app.get('repositories.todo')) as TodoRepository;
  }
}
```

(It would be great if the repository mixin adding `app.repository(TodoRepository)` method was also adding `app.getRepository<TodoRepository>()`, but that's out of scope of this PR.)